opts.go

```diff
@@ -66,3 +69,17 @@ func WithSnapshot(memFilePath, snapshotPath string, opts ...WithSnapshotOpt) Opt
 		m.Handlers.FcInit = loadSnapshotHandlerList
 	}
 }
+
+// MemoryBackendType sets the memory backend type to the given value.
+func MemoryBackendType(typ string) WithSnapshotOpt {
```

Comment:
Please add unit tests specifically for Loadsnapshot in https://github.com/firecracker-microvm/firecracker-go-sdk/blob/main/machine_test.go